testpar/t_select_io_dset.c

```diff
@@ -0,0 +1,3775 @@
+                            if (dtrans) {
+                                if (MAINPROCESS)
+                                    SKIPPED();
+                            }
+                            else
+                                test_multi_dsets_cmpd_with_bkg(fid, chunked, mwbuf);
+
+                            break;
+
+                        case TEST_MULTI_CONV_SIZE_CHANGE: /* case 9 */
+                            if (MAINPROCESS)
+                                TESTING_2("multi-datasets: type conv + size change + no bkg buffer");
+
+                            /* Data transforms does not apply to the dataset datatype for this test */
+                            if (dtrans) {
+                                if (MAINPROCESS)
+                                    SKIPPED();
+                            }
+                            else
+                                test_multi_dsets_size_change_no_bkg(fid, chunked, mwbuf);
+
+                            break;
+
+                        case TEST_MULTI_CONV_SEL_EMPTY: /* case 10 */
+                            if (MAINPROCESS)
+                                TESTING_2("multi-datasets: type conv + empty selections");
+
+                            test_multi_dsets_conv_sel_empty(fid, chunked, dtrans, mwbuf);
+
+                            break;
+
+                        case TEST_MULTI_ALL: /* case 11 */
+                            if (MAINPROCESS)
+                                TESTING_2("multi-datasets: no conv + conv without bkg + conv with bkg");
+
+                            /* Data transforms does not apply to the dataset datatype for this test */
+                            if (dtrans) {
+                                if (MAINPROCESS)
+                                    SKIPPED();
+                            }
+                            else
+                                test_multi_dsets_all(2, fid, chunked, mwbuf);
+
+                            break;
+
+                        case TEST_SELECT_NTESTS:
+                        default:
+                            P_TEST_ERROR;
+                            break;
+
+                    } /* end switch */
+
+                } /* end for test_select_config */
+
+            } /* end mwbuf */
+
+        } /* end dtrans */
+    }     /* end chunked */
+
+    if (H5Pclose(fapl) < 0)
+        P_TEST_ERROR;
+
+    if (H5Fclose(fid) < 0)
+        P_TEST_ERROR;
+
+    if (MAINPROCESS) {
+        printf("\n");
+        TESTING("Testing for H5Pget_no_selection_io_cause()");
+    }
+    test_get_no_selection_io_cause();
+
+    /* Barrier to make sure all ranks are done before deleting the file, and
+     * also to clean up output (make sure PASSED is printed before any of the
+     * following messages) */
+    if (MPI_Barrier(MPI_COMM_WORLD) != MPI_SUCCESS)
+        P_TEST_ERROR;
+
+    /* Delete file */
+    if (MAINPROCESS)
+        if (MPI_File_delete(FILENAME, MPI_INFO_NULL) != MPI_SUCCESS)
```

Comment:
Can this test be run with other parallel-enabled VFDs? If so this should probably be a call to H5Fdelete (either collectively with the file fapl or by MAINPROCESS with a serial fapl) so that VFDs like the Subfiling VFD can correctly delete the test file.